# HG changeset patch
# User Devin J. Pohly <djpohly@gmail.com>
# Date 1343943488 14400
# Node ID ba3512adad88fc0d61ae77fb59dd5c0dd613df8f
# Parent  1e9accbd82a3454e10bdb94ce32a22a15433e817
replace deprecated functions to fix build on ffmpeg 0.11.1

tested on static 0.10.2 as well and it works just fine.

diff -r 1e9accbd82a3 -r ba3512adad88 src/arch/MovieTexture/MovieTexture_FFMpeg.cpp
--- a/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	Mon Aug 06 00:12:23 2012 -0400
+++ b/src/arch/MovieTexture/MovieTexture_FFMpeg.cpp	Thu Aug 02 17:38:08 2012 -0400
@@ -15,6 +15,7 @@
 	{
 		#include <libavformat/avformat.h>
 		#include <libswscale/swscale.h>
+		#include <libavutil/pixdesc.h>
 	}
 };
 
@@ -615,20 +616,23 @@
 	MovieTexture_FFMpeg::RegisterProtocols();
     
 	m_fctx = avcodec::avformat_alloc_context();
+	if( !m_fctx )
+		return "AVCodec: Couldn't allocate context";
     
-    RageFile *f = new RageFile;
-    
-    if( !f->Open(sFile, RageFile::READ) )
-    {
-        RString errorMessage = f->GetError();
-        RString error = ssprintf("MovieDecoder_FFMpeg: Error opening \"%s\": %s", sFile.c_str(), errorMessage.c_str() );
-        delete f;
-        return error;
-    }
-    
-    m_buffer = (unsigned char *)avcodec::av_malloc(STEPMANIA_FFMPEG_BUFFER_SIZE);
-    m_avioContext = avcodec::avio_alloc_context(m_buffer, STEPMANIA_FFMPEG_BUFFER_SIZE, 0, f, AVIORageFile_ReadPacket, NULL, AVIORageFile_Seek);
-    int ret = avcodec::av_open_input_stream( &m_fctx, m_avioContext, sFile.c_str(), NULL, NULL );
+	RageFile *f = new RageFile;
+
+	if( !f->Open(sFile, RageFile::READ) )
+	{
+		RString errorMessage = f->GetError();
+		RString error = ssprintf("MovieDecoder_FFMpeg: Error opening \"%s\": %s", sFile.c_str(), errorMessage.c_str() );
+		delete f;
+		return error;
+	}
+
+	m_buffer = (unsigned char *)avcodec::av_malloc(STEPMANIA_FFMPEG_BUFFER_SIZE);
+	m_avioContext = avcodec::avio_alloc_context(m_buffer, STEPMANIA_FFMPEG_BUFFER_SIZE, 0, f, AVIORageFile_ReadPacket, NULL, AVIORageFile_Seek);
+	m_fctx->pb = m_avioContext;
+	int ret = avcodec::avformat_open_input( &m_fctx, sFile.c_str(), NULL, NULL );
 	if( ret < 0 )
 		return RString( averr_ssprintf(ret, "AVCodec: Couldn't open \"%s\"", sFile.c_str()) );
 
@@ -651,7 +655,7 @@
 		return ssprintf( "AVCodec (%s): %s", sFile.c_str(), sError.c_str() );
 
 	LOG->Trace( "Bitrate: %i", m_pStream->codec->bit_rate );
-	LOG->Trace( "Codec pixel format: %s", avcodec::avcodec_get_pix_fmt_name(m_pStream->codec->pix_fmt) );
+	LOG->Trace( "Codec pixel format: %s", avcodec::av_get_pix_fmt_name(m_pStream->codec->pix_fmt) );
 
 	return RString();
 }
